src/main/java/org/micks/champmaker/championships/ChampionshipController.java

```diff
@@ -21,11 +24,19 @@ public class ChampionshipController {
     @Autowired
     private ChampionshipService championshipService;
 
+    @Autowired
+    private RegisterService registerService;
+
     @GetMapping("/{champId}")
     public ChampionshipDTO getChampionship(@PathVariable long champId) throws EntityNotFoundException {
         return championshipService.getChampionship(champId);
     }
 
+    @PutMapping(value = "/{champId}", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public void registerTeamToChampionship(@PathVariable long champId, @RequestBody RegisterDTO registerDTO) {
```

Comment:
Is there any reason why we don't keep champId inside RegisterDTO?

Reply:
Yes, champId is primary id, inside championshipController. The rest is data for assigning something (team in this case) to championship. I think in classic Rest every PUT request should have path id for resource we are changing.